app/src/main/java/com/example/campusguide/location/CurrentLocation.kt

```diff
@@ -0,0 +1,3 @@
+package com.example.campusguide.location
+
+class CurrentLocation constructor(lat: Double, lon: Double) : Location("Current Location", lat, lon)
```

Comment:
I think you can declare a data class for this.
 https://kotlinlang.org/docs/reference/data-classes.html

Reply:
A data class' primary constructor can only have val / var parameters.
Since we are inheriting from Location which already has properties for lat and lon, turning this into a data class would make the derived class have to override the properties in the base class, resulting in the base class properties being allocated but unused. That seems like a worse design than just not having this be a data class.